### PR TITLE
various conditional flow fixes

### PR DESCRIPTION
**Pull request by gashcrumb**
fixes syndesisio/syndesis#5515
fixes syndesisio/syndesis#5636
fixes syndesisio/syndesis#5608

improves navigation

### CODE REVIEW

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/2307/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Comment by gashcrumb** [comment]
Gonna stick some more stuff on this branch, why not.

File: app/ui-react/packages/api/src/constants.ts [hunk 1]
```diff
@@ -1,12 +1,31 @@
 // Enum for the LeveledMessage level field
 import { Flow } from '@syndesis/models';
+import { Integration } from '@syndesis/models/src';
```
Comment (riccardo-forina) [inline app/ui-react/packages/api/src/constants.ts]: /src

Comment (gashcrumb) [reply app/ui-react/packages/api/src/constants.ts]: damn!  hate that :-)

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/2315/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

File: app/ui-react/packages/api/src/constants.ts (hunk 1 above)
Comment (gashcrumb) [reply app/ui-react/packages/api/src/constants.ts]: ooh. found a bug anyways.

**Comment by riccardo-forina** [comment]
PR Storybook available [here](https://circleci.com/api/v1.1/project/github/syndesisio/syndesis-react/2318/artifacts/0/home/circleci/repo/app/ui-react/doc/index.html)

**Pull request merged**